eWonConnector-gateway/src/main/java/com/hms_networks/americas/sc/ignition/data/SyncDataStateManager.java

```diff
@@ -0,0 +1,496 @@
+   *
+   * @since 1.0.0
+   */
+  private static int failedM2WebExecutionCount = 0;
+
+  /**
+   * The number of successful M2Web metadata thread executions that have occurred since the thread
+   * was started. This value is non-persistent and is reset to zero (0) when the connector is
+   * started/restarted.
+   *
+   * @since 1.0.0
+   */
+  private static int successfulM2WebMetadataExecutionCount = 0;
+
+  /**
+   * The number of failed M2Web metadata thread executions that have occurred since the thread was
+   * started. This value is non-persistent and is reset to zero (0) when the connector is
+   * started/restarted.
+   *
+   * @since 1.0.0
+   */
+  private static int failedM2WebMetadataExecutionCount = 0;
+
+  /**
+   * The number of successful DMWeb thread executions that have occurred since the thread was
+   * started. This value is non-persistent and is reset to zero (0) when the connector is
+   * started/restarted.
+   *
+   * @since 1.0.0
+   */
+  private static int successfulDMWebExecutionCount = 0;
+
+  /**
+   * The number of failed DMWeb thread executions that have occurred since the thread was started.
+   * This value is non-persistent and is reset to zero (0) when the connector is started/restarted.
+   *
+   * @since 1.0.0
+   */
+  private static int failedDMWebExecutionCount = 0;
+
+  /**
+   * Flag indicating if the startup M2Web metadata synchronization has been completed. This value is
+   * non-persistent and is reset to false (false) when the connector is started/restarted.
+   *
+   * @since 1.0.0
+   */
+  private static boolean startupM2WebMetadataSyncCompleted = false;
+
+  /**
+   * Initializes the synchronization data state manager. This method will load existing
+   * synchronization data, or create it if necessary.
+   *
+   * @since 1.0.0
+   */
+  public static void initialize(GatewayContext gatewayContext) {
+    // Store gateway context
+    SyncDataStateManager.gatewayContext = gatewayContext;
+
+    // Reset non-persistent values
+    successfulM2WebExecutionCount = 0;
+    failedM2WebExecutionCount = 0;
+    successfulM2WebMetadataExecutionCount = 0;
+    failedM2WebMetadataExecutionCount = 0;
+    successfulDMWebExecutionCount = 0;
+    failedDMWebExecutionCount = 0;
+    startupM2WebMetadataSyncCompleted = false;
+
+    // Load and store synchronization data, and create it if necessary
+    final long syncDataId = 1L;
+    syncDataState =
+        gatewayContext.getPersistenceInterface().find(EwonSyncDataState.META, syncDataId);
+    if (syncDataState == null) {
+      LOGGER.info("Existing Ewon synchronization data information not found, creating new.");
+      syncDataState =
+          gatewayContext.getLocalPersistenceInterface().createNew(EwonSyncDataState.META);
+      syncDataState.setLong(EwonSyncDataState.ID, syncDataId);
+      gatewayContext.getLocalPersistenceInterface().save(syncDataState);
```

Comment:
throws SException add try/catch

Reply:
Added throws. 

I don't think this class is appropriate for a try/catch. If an exception occurs here, it will be caught by the invoking thread and handled appropriately. In this case, that will result in logging that a critical error occurred and the connector is shutting down. 

If we cannot save values, the connector's behavior cannot be guaranteed (i.e., repeatedly using the same historical transaction ID), so it's best to let the invoking thread handle the critical error. The invoking thread already has access to the gateway context/hook, so its simple for that to perform the shutdown/handling.